### DISCUSSION TITLE
'location' for prefix csv import

### DISCUSSION THREAD

**Discussion by bgoulet00** [nautobot/nautobot#6352] [Q&A] [answered]
i previously experimented with Nautobot 1.x and Netbox and don't recall having any issues importing prefixes but i'm definitely hitting a wall with Nautobot v2.3.6

my location hierarchy :
Site-Type
-Site

in this specific case:
Branch
-USMLB1

the guidance on the import page provides the following:
`A location field could be identified by UUID or by the fields location__name and location__location_type`
there error i'm getting implies it is looking for name and parent though
`Related object not found using the provided attributes: {'name': 'USMLB1', 'parent__name': None}`

this is my data:
location,state,prefix,type
USMLB1,Active,10.1.10.0/24,network

i've tried a ton of variations trying to get this to work
location,state,prefix,type
"USMLB1,Branch",Active,10.1.10.0/24,network
"USMLB1,Site",Active,10.1.10.0/24,network
"location__name:USMLB1,parent__name:Branch",Active,10.1.10.0/24,network
"location__name:USMLB1,location__location_type:Site",Active,10.1.10.0/24,network

i also tried variations with additional columns....
location_name,location_parent,state,prefix,type
USMLB1,Branch,Active,10.1.10.0/24,network

i have gotten it to work with UUID but really need to get it working using site name.   
location,state,prefix,type
ae6bbc3a-7e47-4a63-8b89-7224bfbfa3e7,Active,10.1.10.0/24,network

**Comment by glennmatthews** [comment, marked as answer]
Our docs on this are currently lacking - what it's trying to suggest is something like:

```csv
location__name,status__name,prefix,type
USMLB1,Active,10.1.10.0/24,network
```

or (if you had multiple locations named "USMLB1" that you needed to distinguish between):

```csv
location__name,location__location_type__name,status__name,prefix,type
USMLB1,Branch,Active,10.1.10.0/24,network
```

**Comment by bgoulet00** [comment]
using location_name did allow the import to complete without errors but it didn't actually attach a location to the prefix record.  the resulting prefix has location 'none'

location__name,status,prefix,type
USMLB1,Active,10.1.10.0/24,network

**Reply by glennmatthews** [reply]
hmm, that may be a specific bug actually. The `location` field is actually an alias to the many-to-many `locations` field and it may be that the CSV import isn't handling that special case correctly.

**Reply by bgoulet00** [reply]
i'm having a similar issue with assigning vlans to prefixes.
vlan=UUID works
vlan_vid=2 imports without error but results in no vlan assignment

Nautobot version: 2.3.6
Django version: 4.2.16

**Reply by bgoulet00** [reply]
if you think this may be a bug, is there some bug submission i should do?

**Reply by glennmatthews** [reply]
yes, sorry - would you mind opening an issue at https://github.com/nautobot/nautobot/issues/new?assignees=&labels=type%3A+bug%2Ctriage&projects=&template=bug_report.md with the specific non-working data in both of these cases? One issue will probably cover both.

**Reply by bgoulet00** [reply]
looks like i was doing something wrong.  `location_name` (single underscore) is what i was doing and imports without error but doesn't actually associate the prefix to the site.  i just realized your example `location__name`  is double underscore.  using that the import works
